src/app/details/details.component.ts

```diff
@@ -9,9 +11,45 @@ import { CityWeatherInfo } from '../shared/interfaces/city-weather-info.interfac
 })
 export class DetailsComponent {
   weatherInfo: CityWeatherInfo = {
-    city: 'Kyiv, Ukraine',
-    date: 'February 7th, 2022',
-    temp: '7 °С',
-    isFavorite: false
+    city: 'Kiev, Ukraine',
+    date: '',
+    temp: '',
+    isFavorite: false,
+  };
+
+  position: any = {};
+
+  constructor(private weatherServise: WeatherService) {}
+
+  async ngOnInit() {
+    if (navigator.geolocation) {
```

Comment:
this.geolocationService.getPosition()
.pipe(
  swithMap(({ area }) => {
    return this.weatherServise.getCurrentWeatherByArea(area )
  })
)
.subscribe((weather => {
  this.weatherInfo = weather;
}))

Reply:
getCurrentWeather() {
reuturn this.weatherAPI.getweathe().pipe(
   swithMap
   ...
   map()
)
}


homeService.getCurrentWeather()
.subscribe()